app/assets/src/components/views/SampleUploadFlow/utils.js

```diff
@@ -19,3 +21,35 @@ export const openBasespaceOAuthPopup = params => {
     BASESPACE_OAUTH_WINDOW_HEIGHT
   );
 };
+
+// The following three functions were extracted from SampleTypeSearchBox.
+// They aid client-side search of options.
+export const matchType = (type, query) => {
```

Comment:
Good call. I renamed `type` to `result` everywhere.

> I wonder if components/utils/search would be a better place for these functions

I don't see any `util.js` in there and it would see more strange to have it in a source tree where it is not used. For re-use, I expect a developer will first want something "like the upload page" then find the functions. At that point, then should reorganize as they see fit. 

As a general comment, it seems strange to me that there are so many util.js files in our dirs but so few global js utils.